test/grpc/unary.cc

```diff
@@ -19,7 +19,8 @@ using helloworld::HelloRequest;
 
 using stout::Borrowable;
 
-TEST(UnaryTest, Success) {
+void TestUnaryWithClient(
+    std::function<Client(Borrowable<CompletionPool>&&, int)> client_factory) {
```

Comment:
When declaring a function which takes an `std::function` which it only calls *but does not store*, prefer to take it by `const &`.

std::function can be ~heavy to copy. (Think of it like a ~big functor struct)

Reply:
> std::function can be ~heavy to copy.
What about passing it as rvalue?

Reply:
Why? (What's the advantage over passing by const ref, where there's no requirement to construct or copy an object at all?)

Reply:
👍🏻